packages/@aws-cdk/aws-stepfunctions-tasks/lib/apigateway/invoke-http-api.ts

```diff
@@ -0,0 +1,62 @@
+import * as apigatewayv2 from '@aws-cdk/aws-apigatewayv2';
+import * as iam from '@aws-cdk/aws-iam';
+import * as sfn from '@aws-cdk/aws-stepfunctions';
+import * as cdk from '@aws-cdk/core';
+import { Construct } from 'constructs';
+import { InvokeApiGatewayApiBase } from './base';
+import { InvokeApiGatewayApiBaseProps } from './base-types';
+
+/**
+ * Properties for invoking an HTTP API Endpoint
+ */
+export interface InvokeApiGatewayHttpApiProps extends InvokeApiGatewayApiBaseProps {
+  /**
+   * API to call
+   */
+  readonly api: apigatewayv2.IHttpApi;
+
+  /**
+   * Name of the stage where the API is deployed to in API Gateway
+   * @default '$default'
+   */
+  readonly stageName?: string;
+}
+
+/**
+ * Invoke HTTP API endpoint as a Task
+ *
+ * @see https://docs.aws.amazon.com/step-functions/latest/dg/connect-api-gateway.html
+ */
+export class InvokeApiGatewayHttpApi extends InvokeApiGatewayApiBase {
```

Comment:
We use the word `Invoke` in the context of lambda because that's intuitive to lambda and is the official term for its operation. Should we find a different word here, say `Call`? (This is what the [official docs](https://docs.aws.amazon.com/step-functions/latest/dg/connect-api-gateway.html) call it)

Why have you chosen to separate these into two different classes? How would it be if this was modified to a single class, say `CallApiGatewayEndpoint`, which takes two optional properties of types `IHttpApi` and `IRestApi` and enforces one, and only one, of them is provided?

Reply:
1. I agree on using `Call` instead of `Invoke`.
2. I implemented it this way to make the separation explicit so the if in future there is a way to call WebSocket API as well (which I believe might have different way compared to these two), we have clear separation and callers will not get confused. Given that, I am still okay with implementing a single class with validations around the passed props. Let me know your thoughts @nija-at @shivlaks

Reply:
3. Another point for separation could be that some props (eg. stageName) can be optional for one but required for other. Also, some prop might not even be supported for one of them.